File: app/src/main/java/com/kota/Telnet/TelnetStateHandler.java
package com.kota.Telnet;

import com.kota.Telnet.Model.TelnetRow;

import java.util.Vector;

public abstract class TelnetStateHandler {
    int _current_page = 0;

    public abstract void handleState();

    public void cleanRow(int row) {
        if (TelnetClient.getModel() != null) {
            TelnetClient.getModel().cleanRow(row);
        }
    }

    public void cleanFrame() {
        if (TelnetClient.getModel() != null) {
            TelnetClient.getModel().cleanFrame();
        }
    }

    public String getRowString(int row) {
        if (TelnetClient.getModel() != null) {
            return TelnetClient.getModel().getRowString(row);
        }
        return "";
    }

    public Vector<TelnetRow> getRows() {
        return TelnetClient.getModel().getRows();
    }

    public void setCurrentPage(int pageID) {
        this._current_page = pageID;
    }

    public int getCurrentPage() {
        return this._current_page;
    }

    public void clear() {
    }
}

File: app/src/main/java/com/kota/Telnet/TelnetStateHandler.kt
package com.kota.Telnet

import com.kota.Telnet.Model.TelnetModel.cleanFrame
import com.kota.Telnet.Model.TelnetModel.cleanRow
import com.kota.Telnet.Model.TelnetModel.getRowString
import com.kota.Telnet.Model.TelnetModel.rows
import com.kota.Telnet.Model.TelnetRow

abstract class TelnetStateHandler {
    var currentPage: Int = 0

    abstract fun handleState()

    fun cleanRow(row: Int) {
        if (TelnetClient.Companion.getModel() != null) {
            TelnetClient.Companion.getModel().cleanRow(row)
        }
    }

    fun cleanFrame() {
        if (TelnetClient.Companion.getModel() != null) {
            TelnetClient.Companion.getModel().cleanFrame()
        }
    }

    fun getRowString(row: Int): String {
        if (TelnetClient.Companion.getModel() != null) {
            return TelnetClient.Companion.getModel().getRowString(row)
        }
        return ""
    }

    val rows: Vector<TelnetRow?>
        get() = TelnetClient.Companion.getModel().rows

    open fun clear() {
    }
}
